### DISCUSSION TITLE
Explicit Valence Error for PubChemQC Molecules

### DISCUSSION THREAD

**Discussion by Prometheus1400** [rdkit/rdkit#4379] [Q&A] [answered]
I am interested in doing some deep learning tasks using the PubChemQC dataset. I am trying to generate molecules from the ```.mol``` files which contain the 3D coordinates. However, I keep getting the error "Explicit valence for atom X is greater than permitted", where X is usually Boron or Nitrogen. From my understanding this should indicate that the molecule is not chemically valid, but why would this be the case from a reputable dataset?  

This is the contents of a usual ```.mol``` file, and in this case one that generates the explicit valence error.
```.txt
PUBCHEM 074890404 B3LYP 6-31G(d) gamess geometry optimization
 OpenBabel08231421183D

 19 19  0  0  0  0  0  0  0  0999 V2000
    0.9117    0.0338    0.1977 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.3732   -1.2236   -0.3710 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.0799    0.0045   -0.3980 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.0216   -1.1842   -0.1757 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.4614    1.2238   -0.2353 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.0199    1.2447   -0.0275 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.2240    2.6194   -0.2629 B   0  0  0  0  0  0  0  0  0  0  0  0
    2.3614   -0.0032   -0.0108 N   0  0  0  0  0  0  0  0  0  0  0  0
    2.3283    2.2657    0.1338 O   0  0  0  0  0  0  0  0  0  0  0  0
    4.5436    3.7764   -0.0936 O   0  0  0  0  0  0  0  0  0  0  0  0
    6.5806    2.7123   -0.4592 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.5245   -0.9862    0.1844 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.4397    0.6261   -0.5896 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.6873    0.5091    1.1555 H   0  0  0  0  0  0  0  0  0  0  0  0
    4.8753   -2.1750   -0.4996 H   0  0  0  0  0  0  0  0  0  0  0  0
    6.1574   -0.0396   -0.5550 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.4069   -2.0774   -0.1416 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.5933    3.5627    0.0366 H   0  0  0  0  0  0  0  0  0  0  0  0
    7.0141    1.8579   -0.5725 H   0  0  0  0  0  0  0  0  0  0  0  0
  1 14  1  0  0  0  0
  2  4  2  0  0  0  0
  3  2  1  0  0  0  0
  3  5  1  0  0  0  0
  4 17  1  0  0  0  0
  4  8  1  0  0  0  0
  5  6  1  0  0  0  0
  6  8  1  0  0  0  0
  6  9  2  0  0  0  0
  7  5  2  0  0  0  0
  7 10  1  0  0  0  0
  8  1  1  0  0  0  0
 10 18  1  0  0  0  0
 11  7  1  0  0  0  0
 12  1  1  0  0  0  0
 13  1  1  0  0  0  0
 15  2  1  0  0  0  0
 16  3  1  0  0  0  0
 19 11  1  0  0  0  0
M  RAD  1   3   2
M  END
```
Any help would be appreciated.

**Comment by greglandrum** [comment, marked as answer]
As you mentioned, that type of error almost always indicates that the molecule is not chemically reasonable.
In the case of the molecule you include that is certainly the case, boron should not have a valence of four like this:
![image](https://user-images.githubusercontent.com/540511/128117598-2d34d4b5-947b-4c8a-81f9-6fabf6d3a2e5.png)

I can't explain why molecules like this are in the PubchemQC dataset, that's a question best answered by the people who created the dataset.

**Reply by Prometheus1400** [reply]
Thanks for the reply Greg, I will have to look into this further with the PubChem people, cheers!
